README.md

```diff
@@ -1,4 +1,199 @@
+foreach (range(30, 40) as $age) {
+    $person = $repository->createRow();
+    $person->name = 'John';
+    $person->surname = 'Doe';
+    $person->age = $age;
+    $persons[] = $person;
+}
+$repository->insert($persons);
+```
+
+#### Updating an entity
+
+```php
+$entity = $repository->find($id);
+$entity->name = 'Jake';
+$entity->save();
+```
+
+Classic:
+```php
+$repository->update($id, [
+    Person::name => 'Jake',
+]);
+```
+
+#### Deleting an entity
+
+```php
+$entity = $repository->find($id);
+$entity->delete();
+```
+
+Or:
+```php
+$repository->delete($id);
+```
+### Scopes
+Scope is a way to define which behaviors are enabled for repository, query and entity recursively.
+
+Recursively means, that if you define a scope for repository, it will be applied to query and entity as well.
+
+`->scopeRaw()` returns a clone of the object with raw scope. Raw scope removes all behaviors.
+
+`->scopeFull()` returns a clone of the object with full scope. Full scope keeps all behaviors.
+
+#### Example
+```php
+final class ApiScope implements \Efabrica\NetteRepository\Repository\Scope\Scope
+{
+    public function apply(RepositoryBehaviors $behaviors, Repository $repository): void
+    {
+        $behaviors
+            ->remove(SoftDeleteBehavior::class)
+            ->remove(PublishBehavior::class);
+        // You should only remove behaviors, as the scope can be applied on different repositories at once.
+        
+        // If you had a user repository and injected the user into the scope, you could do this:
+        if ($repository instanceof UserRepository && $this->user->isAdmin()) {
+            $behaviors->add(AdminBehavior::class); // This behavior does not exist, it's just an example.
+        }
+    }
+}
+```
+
+You should then create your own ScopeContainer:
+
+```php
+abstract class RepositoryBase extends \Efabrica\NetteRepository\Repository\Repository
+{
+    /** @inject */
+    public ApiScope $apiScope;
+    
+    public function scopeApi(): self
+    {
+        return $this->withScope($this->apiScope);
+    }
+}
+```
+
+And implement shorthand methods for your queries:
+
+```php
+abstract class QueryBase extends \Efabrica\NetteRepository\Repository\Query
+{
+    public function scopeApi(): self
+    {
+        // returns cloned query with Api scope set
+        return $this->withScope($this->repository->apiScope);
+        // or if api scope has no dependencies, you can use this:
+        return $this->withScope(new ApiScope());
+    }
+}
+```
+
+Usage:
+```php
+// all of these are equivalent:
+$repository->findBy(['age > ?' => 18])->scopeApi()->fetchAll();
+$repository->scopeApi()->findBy(['age > ?' => 18])->fetchAll();
+$repository->query()->where('age > ?', 18)->scopeApi()->fetchAll();
+$repository->scopeApi()->query()->where('age > ?', 18)->fetchAll();
+$repository->query()->scopeApi()->where('age > ?', 18)->fetchAll();
+```
+
+## Code Generator
+
+Code generation is fully optional, but it is recommended to use it. 
```

Comment:
Not quite clear how to turn on or turn off the code generation, if you need to run a specific command, etc.